### PR TITLE
[CARBONDATA-718] Add rat to check if all files contain ASF header and…

### PR DESCRIPTION
**Pull request by jbonofre**
… fix files with missing ASF header

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[CARBONDATA-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).
 - [X] Testing done
 
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - What manual testing you have done?
        - Any additional information to help reviewers in testing this change.
         
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
                 
---

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @chenliang613

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/970/

File: pom.xml [hunk 1]
```diff
@@ -241,6 +241,32 @@
     </pluginManagement>
     <plugins>
       <plugin>
+        <groupId>org.apache.rat</groupId>
+        <artifactId>apache-rat-plugin</artifactId>
```
Comment (chenliang613) [inline pom.xml]: In main pom.xml , already exists rat plugin , can you just update it:
<plugin>
            <groupId>org.apache.rat</groupId>
            <artifactId>apache-rat-plugin</artifactId>
            <version>0.11</version>
            <executions>
              <execution>
                <phase>verify</phase>
                <goals>
                  <goal>check</goal>
                </goals>
              </execution>
            </executions>
            <configuration>
              <reportFile>${project.build.directory}/${project.build.finalName}.rat</reportFile>
              <excludeSubProjects>false</excludeSubProjects>
              <useDefaultExcludes>true</useDefaultExcludes>
              <excludes>
                <exclude>**/*.iml</exclude>
                <exclude>**/*.csv</exclude>
                <exclude>**/*.dictionary</exclude>
                <exclude>**/*.ktr</exclude>
                <exclude>**/*.rat</exclude>
                <exclude>**/_SUCCESS</exclude>
                <exclude>**/non-csv</exclude>
                <exclude>**/.invisibilityfile</exclude>
                <exclude>**/noneCsvFormat.cs</exclude>
                <exclude>**/org.apache.spark.sql.sources.DataSourceRegister</exclude>
                <exclude>**/org.apache.spark.sql.test.TestQueryExecutorRegister</exclude>
                <exclude>**/derby.log</exclude>
                <exclude>**/meta.lock</exclude>
                <exclude>**/loadmetadata.metadata</exclude>
                <exclude>**/modifiedTime.mdt</exclude>
                <exclude>**/PULL_REQUEST_TEMPLATE.md</exclude>
                <exclude>**/dict.txt</exclude>
                <exclude>**/sample</exclude>
                <exclude>**/test.json</exclude>
              </excludes>
            </configuration>
          </plugin>

Comment (jbonofre) [inline pom.xml]: Good catch. I'm fixing.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/972/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**